clang/lib/CIR/Dialect/IR/CIRTypes.cpp

```diff
@@ -403,6 +403,30 @@ uint64_t PointerType::getPreferredAlignment(
   return 8;
 }
 
+llvm::TypeSize
+DataMemberType::getTypeSizeInBits(const ::mlir::DataLayout &dataLayout,
+                                  ::mlir::DataLayoutEntryListRef params) const {
+  // FIXME: consider size differences under different ABIs
```

Comment:
Please make these two FIXME's into their own Unimplemented features and add asserts here.

Reply:
The header file `UnimplementedFeatureGuarding.h` is located under `CIR/CodeGen` and this file is located under `CIR/Dialect`. I'm not sure whether it is a good idea to add a guard in the `UnimplementedFeature` class and use it here? Maybe we should move the `UnimplementedFeatureGuarding.h` header to a more common place in its parent directories?

Reply:
Good catch! We probably need a separate feature guarding header for dialect stuff, so we could have one for CIRGen and another for CIR dialect (maybe use a slightly different name for the header?). Anyways, content for a follow up PR (if you want to work on that, great).

Reply:
Sounds good. I'll make a follow up PR on this.